src/plugins/telemetryTable/TelemetryTableViewProvider.js

```diff
@@ -19,100 +19,103 @@
  * this source code distribution or the Licensing information page available
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
+import TableComponent from './components/table.vue';
+import TelemetryTable from './TelemetryTable';
+import Vue from 'vue';
 
-define([
-    './components/table.vue',
-    './TelemetryTable',
-    'vue'
-], function (
-    TableComponent,
-    TelemetryTable,
-    Vue
-) {
-    function TelemetryTableViewProvider(openmct) {
-        function hasTelemetry(domainObject) {
-            if (!Object.prototype.hasOwnProperty.call(domainObject, 'telemetry')) {
-                return false;
-            }
+class TelemetryTableView {
+    constructor(openmct, domainObject, objectPath) {
+        this.openmct = openmct;
+        this.domainObject = domainObject;
+        this.objectPath = objectPath;
+        this.component = undefined;
 
-            let metadata = openmct.telemetry.getMetadata(domainObject);
+        this.table = new TelemetryTable(domainObject, openmct);
+    }
 
-            return metadata.values().length > 0;
+    getViewContext() {
+        if (this.component) {
+            return this.component.$refs.tableComponent.getViewContext();
+        } else {
+            return {
+                type: 'telemetry-table'
```

Comment:
What does this do?

Reply:
fixed, I think was copied from old code.